imap/mboxlist.c

```diff
@@ -1895,22 +1895,32 @@ EXPORTED int mboxlist_createmailbox(const mbentry_t *mbentry,
     else if (config_getswitch(IMAPOPT_MAILBOX_LEGACY_DIRS))
         mbtype |= MBTYPE_LEGACY_DIRS;
 
+    newmbentry = mboxlist_entry_create();
+    newmbentry->acl = xstrdupnull(acl);
+    newmbentry->mbtype = mbtype | MBTYPE_RESERVE;
```

Comment:
Does the RESERVE flag later get removed?

Reply:
Yes.  We do `newmbentry->mbtype = mbtype` below after the mailbox has been created.